src/RcppExports.cpp

```diff
@@ -428,6 +428,127 @@ BEGIN_RCPP
+    Rcpp::RNGScope rcpp_rngScope_gen;
+    Rcpp::traits::input_parameter< NumericVector >::type iC(iCSEXP);
+    Rcpp::traits::input_parameter< double >::type v(vSEXP);
+    Rcpp::traits::input_parameter< double >::type tStep(tStepSEXP);
+    rcpp_result_gen = Rcpp::wrap(iCones2Cells_rcpp(iC, v, tStep));
+    return rcpp_result_gen;
+END_RCPP
+}
+// blockedAngle_rcpp
+NumericVector blockedAngle_rcpp(int n, List state, NumericMatrix p_pred, List objects);
+RcppExport SEXP _m4ma_blockedAngle_rcpp(SEXP nSEXP, SEXP stateSEXP, SEXP p_predSEXP, SEXP objectsSEXP) {
+BEGIN_RCPP
+    Rcpp::RObject rcpp_result_gen;
+    Rcpp::RNGScope rcpp_rngScope_gen;
+    Rcpp::traits::input_parameter< int >::type n(nSEXP);
+    Rcpp::traits::input_parameter< List >::type state(stateSEXP);
+    Rcpp::traits::input_parameter< NumericMatrix >::type p_pred(p_predSEXP);
+    Rcpp::traits::input_parameter< List >::type objects(objectsSEXP);
+    rcpp_result_gen = Rcpp::wrap(blockedAngle_rcpp(n, state, p_pred, objects));
+    return rcpp_result_gen;
+END_RCPP
+}
+// getLeaders_rcpp
+Nullable<List> getLeaders_rcpp(int n, List state, NumericMatrix centres, List objects, bool onlyGroup, bool preferGroup, bool pickBest);
+RcppExport SEXP _m4ma_getLeaders_rcpp(SEXP nSEXP, SEXP stateSEXP, SEXP centresSEXP, SEXP objectsSEXP, SEXP onlyGroupSEXP, SEXP preferGroupSEXP, SEXP pickBestSEXP) {
+BEGIN_RCPP
+    Rcpp::RObject rcpp_result_gen;
+    Rcpp::RNGScope rcpp_rngScope_gen;
+    Rcpp::traits::input_parameter< int >::type n(nSEXP);
+    Rcpp::traits::input_parameter< List >::type state(stateSEXP);
+    Rcpp::traits::input_parameter< NumericMatrix >::type centres(centresSEXP);
+    Rcpp::traits::input_parameter< List >::type objects(objectsSEXP);
+    Rcpp::traits::input_parameter< bool >::type onlyGroup(onlyGroupSEXP);
+    Rcpp::traits::input_parameter< bool >::type preferGroup(preferGroupSEXP);
+    Rcpp::traits::input_parameter< bool >::type pickBest(pickBestSEXP);
+    rcpp_result_gen = Rcpp::wrap(getLeaders_rcpp(n, state, centres, objects, onlyGroup, preferGroup, pickBest));
+    return rcpp_result_gen;
+END_RCPP
+}
+// getBuddy_rcpp
+Nullable<List> getBuddy_rcpp(int n, NumericVector group, NumericVector a, NumericMatrix p_pred, NumericMatrix centres, List objects, bool pickBest, List state);
+RcppExport SEXP _m4ma_getBuddy_rcpp(SEXP nSEXP, SEXP groupSEXP, SEXP aSEXP, SEXP p_predSEXP, SEXP centresSEXP, SEXP objectsSEXP, SEXP pickBestSEXP, SEXP stateSEXP) {
+BEGIN_RCPP
+    Rcpp::RObject rcpp_result_gen;
+    Rcpp::RNGScope rcpp_rngScope_gen;
+    Rcpp::traits::input_parameter< int >::type n(nSEXP);
+    Rcpp::traits::input_parameter< NumericVector >::type group(groupSEXP);
+    Rcpp::traits::input_parameter< NumericVector >::type a(aSEXP);
+    Rcpp::traits::input_parameter< NumericMatrix >::type p_pred(p_predSEXP);
+    Rcpp::traits::input_parameter< NumericMatrix >::type centres(centresSEXP);
+    Rcpp::traits::input_parameter< List >::type objects(objectsSEXP);

```

Comment:
the conversion between c++ and R you do here is interesting, and reminds me to ask you: did you try to parallelise the msumloglike? If you recall, I did try myself but I couldn't overcome the conversion between R and c++ for the type list() which contained many different types of structure (something that c++ didn't like at all), and it was not comparable to the Rcpp equivalent: List, which indeed accepted only one type.  However, I was not aware of the wrap() functionality and of the Rcpp::traits::input_parameter.

Reply:
Yes, I recently discovered this [example](https://gallery.rcpp.org/articles/custom-as-and-wrap-example/) showing how to write wrappers for Rcpp and regular C++ objects. This might indeed help with the parallelization. I will make a new task for this! Nlesc-team/Sushi#288